cmd/kubectl-gadget/utils/trace.go

```diff
@@ -465,52 +468,77 @@ func getTraceWatcher(traceID string) (watch.Interface, error) {
 func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Trace) bool) (*gadgetv1alpha1.TraceList, error) {
 	var returnedTraces gadgetv1alpha1.TraceList
 
-	tracesNumber := 0
-	watchedTracesNumber := 0
+	traceList, err := getTraceListFromID(traceID)
+	if err != nil {
+		return nil, err
+	}
+
+	tracesNumber := len(traceList.Items)
+	tracesReady := 0
+
+	// Maybe some traces are already ready?
+	for _, trace := range traceList.Items {
+		if conditionFunction(&trace) {
+			tracesReady++
+		}
+	}
```

Comment:
You are right, the logic is wrong here.
I wrote it a bite late and just figured out we can have some traces ready and some not instead of either all ready or none ready.